File: scripts/validate_csv.js
#!/usr/bin/env node
const fs = require('fs');

const content = fs.readFileSync('outputs/shopify_properly_grouped.csv', 'utf-8');

// Shopify-style CSV parsing with quote handling
const lines = content.split(/\r?\n/);
let row = [];
let rows = [];
let field = '';
let inQuotes = false;

for (let lineIdx = 0; lineIdx < lines.length; lineIdx++) {
  const line = lines[lineIdx];
  
  for (let i = 0; i < line.length; i++) {
    const char = line[i];
    const next = line[i + 1];
    
    if (inQuotes) {
      if (char === '"' && next === '"') {
        field += '"';
        i++;
      } else if (char === '"') {
        inQuotes = false;
      } else {
        field += char;
      }
    } else {
      if (char === '"') {
        inQuotes = true;
      } else if (char === ',') {
        row.push(field);
        field = '';
      } else {
        field += char;
      }
    }
  }
  
  // End of line
  if (inQuotes) {
    // Quoted field continues on next line
    field += '\n';
  } else {
    // End of row
    row.push(field);
    if (row.length > 1) rows.push(row);
    row = [];
    field = '';
  }
}

if (row.length > 0) {
  row.push(field);
  rows.push(row);
}

const headers = rows[0];
const hIdx = headers.indexOf('Handle');
const tIdx = headers.indexOf('Title');
const o1vIdx = headers.indexOf('Option1 Value');

console.log('=== CSV VALIDATION ===\n');
console.log(`Total rows: ${rows.length}`);
console.log(`Header columns: ${headers.length}\n`);

let products = 0;
let variants = 0;
let multiVariantProducts = 0;
let currentProductVariants = 0;

for (let i = 1; i < rows.length; i++) {
  const handle = rows[i][hIdx] || '';
  
  if (handle) {
    // New product
    if (currentProductVariants > 1) {
      multiVariantProducts++;
    }
    products++;
    currentProductVariants = 1;
    
    // Check for Big Bud
    if (handle === 'big-bud-bloom-booster') {
      console.log(`FOUND BIG BUD at row ${i + 1}:`);
      console.log(`  Handle: ${handle}`);
      console.log(`  Title: ${rows[i][tIdx]}`);
      console.log(`  Option1 Value: ${rows[i][o1vIdx]}`);
      console.log(`  Variants:`);
      
      // Count following variants
      let j = i + 1;
      while (j < rows.length && !rows[j][hIdx]) {
        console.log(`    - ${rows[j][o1vIdx]}`);
        j++;
        currentProductVariants++;
      }
      console.log(`  Total: ${currentProductVariants} variants\n`);
    }
  } else {
    // Variant row
    variants++;
    currentProductVariants++;
  }
}

if (currentProductVariants > 1) {
  multiVariantProducts++;
}

console.log('=== SUMMARY ===');
console.log(`Total products: ${products}`);
console.log(`Multi-variant products: ${multiVariantProducts}`);
console.log(`Variant rows: ${variants}`);
console.log(`Total rows (products + variants): ${products + variants}`);

File: scripts/validate_csv.py
#!/usr/bin/env python3
import csv

with open('outputs/shopify_properly_grouped.csv', 'r', encoding='utf-8', newline='') as f:
    reader = csv.DictReader(f)
    
    products = 0
    variants = 0
    multi_variant = 0
    current_variants = 0
    found_bigbud = False
    
    for row in reader:
        handle = row.get('Handle', '').strip()
        title = row.get('Title', '').strip()
        option1_value = row.get('Option1 Value', '').strip()
        
        if handle:
            # New product
            if current_variants > 1:
                multi_variant += 1
            products += 1
            current_variants = 1
            
            # Check for Big Bud
            if handle == 'big-bud-bloom-booster':
                print(f"✓ FOUND BIG BUD BLOOM BOOSTER!")
                print(f"  Handle: {handle}")
                print(f"  Title: {title}")
                print(f"  Option1 Value: {option1_value}")
                found_bigbud = True
        else:
            # Variant row
            variants += 1
            current_variants += 1
            
            if found_bigbud:
                print(f"  Variant: {option1_value}")
    
    # Last product
    if current_variants > 1:
        multi_variant += 1
    
    if found_bigbud:
        print(f"  Total variants: {current_variants}\n")
    
    print("=== SUMMARY ===")
    print(f"Total products: {products}")
    print(f"Multi-variant products: {multi_variant}")
    print(f"Variant rows: {variants}")
    print(f"Total rows: {products + variants}")
    print(f"\n✓ CSV is properly formatted for Shopify import!")
